presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoInstaller.java

```diff
@@ -13,66 +13,74 @@
-            nodeps = "--nodeps";
-        }
-        String installRpm = format("sudo rpm -i %s %s", nodeps, pathToRpm);
-        if (executeCommand(installRpm) != 0) {
-            throw new PrestoManagerException("Failed to install presto");
-        }
+        this.packageType = requireNonNull(packageType);
+        this.urlToFetchPackage = requireNonNull(urlToFetchPackage);
+        this.disableDependencyChecking = requireNonNull(disableDependencyChecking);
     }
 
-    public static void uninstallPresto(String packageName, boolean disableDependencyChecking, boolean ignoreNotInstalled)
+    public void runCommand()
             throws PrestoManagerException
     {
-        String packageInstalled = format("sudo rpm -q %s", packageName);
-        if (executeCommand(packageInstalled) != 0) {
-            if (!ignoreNotInstalled) {
-                throw new PrestoManagerException(format("Package %s is not installed", packageName));
-            }
-        }
-        else {
-            String nodeps = "";
-            if (disableDependencyChecking) {
-                nodeps = "--nodeps";
-            }
-            String uninstallPackage = format("sudo rpm -e %s %s", nodeps, packageName);
-            if (executeCommand(uninstallPackage) != 0) {
-                throw new PrestoManagerException(format("Failed to uninstall package %s", packageName));
-            }
+        switch (packageType) {
+            case RPM:
+                File tempFile;
+                try {
+                    tempFile = createTempFile("presto", ".rpm");
+                    copyURLToFile(urlToFetchPackage, tempFile, CONNECTION_TIMEOUT, READ_TIMEOUT);
+                }
+                catch (IOException e) {
+                    throw new PrestoManagerException(format("Failed to download file from url %s", urlToFetchPackage), e);
+                }
+                installUsingRpm(tempFile.toString(), disableDependencyChecking);
+                tempFile.delete();
+                break;
+            case TARBALL:
+                // TODO: Add tarball installation
+                throw new UnsupportedOperationException("Tarball installation is not supported");
+            default:
+                // TODO: Add to logger
+                throw new IllegalArgumentException(format("Unsupported package type %s", packageType));
         }
     }
 
-    public static void upgradePresto(String pathToRpm, boolean disableDependencyChecking)
+    private static void installUsingRpm(String pathToRpm, boolean disableDependencyChecking)
             throws PrestoManagerException
     {
-        String checkRpm = format("rpm -Kv --nosignature %s", pathToRpm);
+        String checkRpm = format("rpm -K --nosignature %s", pathToRpm);
         if (executeCommand(checkRpm) != 0) {
             throw new PrestoManagerException("Corrupted RPM");
         }
         String nodeps = "";
         if (disableDependencyChecking) {
             nodeps = "--nodeps";
         }
```

Comment:
I believe its not a good idea to use ternary operators. It makes the code difficult to understand.